---
title: "ADH and ALDH gene regulation"
author: "Evgeniia Golovina"
date: "03/02/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# install.packages("pacman")
# load libraries
pacman::p_load(ggplot2, ComplexHeatmap, dplyr, tidyr, reshape2, tidyverse, circlize, LDlinkR)

setwd("/Users/foffa/Desktop/ADH_and_ALDH_regulatory_networks")
```

## Reporducibility report for "*ADH* and *ALDH* gene regulation" study.

This R Markdown document generates the reproducibility report for "*ADH* and *ALDH* gene regulation" study. 

### 1. CoDeS3D analysis

First, we run CoDeS3D pipeline (gene mode, GTEx eqtl project) to get regulatory interactions for each of *ADH* and *ALDH* gene across all GTEx (v8) tissues: `python codes3d/codes3d.py -g ALDH1A1 -o results/codes3d/ALDH1A1 --eqtl-project GTEx`. 

```{r loading_codes3d_results}
# 6 ADH genes; there were no significant interactions for ADH7 gene
ADH1A <- read.table("results/codes3d/ADH1A/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH1B <- read.table("results/codes3d/ADH1B/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH1C <- read.table("results/codes3d/ADH1C/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH4 <- read.table("results/codes3d/ADH4/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH5 <- read.table("results/codes3d/ADH5/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH6 <- read.table("results/codes3d/ADH6/significant_eqtls.txt", header = TRUE, sep = "\t")
# 17 ALDH genes; there were no ignificant interactions for ALDH8A1 and ALDH9A1 genes
ALDH16A1 <- read.table("results/codes3d/ALDH16A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH18A1 <- read.table("results/codes3d/ALDH18A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A1 <- read.table("results/codes3d/ALDH1A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A2 <- read.table("results/codes3d/ALDH1A2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A3 <- read.table("results/codes3d/ALDH1A3/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1B1 <- read.table("results/codes3d/ALDH1B1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1L1 <- read.table("results/codes3d/ALDH1L1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1L2 <- read.table("results/codes3d/ALDH1L2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH2 <- read.table("results/codes3d/ALDH2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3A1 <- read.table("results/codes3d/ALDH3A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3A2 <- read.table("results/codes3d/ALDH3A2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3B1 <- read.table("results/codes3d/ALDH3B1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3B2 <- read.table("results/codes3d/ALDH3B2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH4A1 <- read.table("results/codes3d/ALDH4A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH5A1 <- read.table("results/codes3d/ALDH5A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH6A1 <- read.table("results/codes3d/ALDH6A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH7A1 <- read.table("results/codes3d/ALDH7A1/significant_eqtls.txt", header = TRUE, sep = "\t")

# remove interactions with missing rsID number for snp
# 6 ADH genes
ADH1A <- ADH1A[ADH1A$snp != ".", ]; ADH1B <- ADH1B[ADH1B$snp != ".", ]
ADH1C <- ADH1C[ADH1C$snp != ".", ]; ADH4 <- ADH4[ADH4$snp != ".", ]
ADH5 <- ADH5[ADH5$snp != ".", ]; ADH6 <- ADH6[ADH6$snp != ".", ]
# 17 ALDH genes
ALDH16A1 <- ALDH16A1[ALDH16A1$snp != ".", ]; ALDH18A1 <- ALDH18A1[ALDH18A1$snp != ".", ]
ALDH1A1 <- ALDH1A1[ALDH1A1$snp != ".", ]; ALDH1A2 <- ALDH1A2[ALDH1A2$snp != ".", ]
ALDH1A3 <- ALDH1A3[ALDH1A3$snp != ".", ]; ALDH1B1 <- ALDH1B1[ALDH1B1$snp != ".", ]
ALDH1L1 <- ALDH1L1[ALDH1L1$snp != ".", ]; ALDH1L2 <- ALDH1L2[ALDH1L2$snp != ".", ]
ALDH2 <- ALDH2[ALDH2$snp != ".", ]; ALDH3A1 <- ALDH3A1[ALDH3A1$snp != ".", ]
ALDH3A2 <- ALDH3A2[ALDH3A2$snp != ".", ]; ALDH3B1 <- ALDH3B1[ALDH3B1$snp != ".", ]
ALDH3B2 <- ALDH3B2[ALDH3B2$snp != ".", ]; ALDH4A1 <- ALDH4A1[ALDH4A1$snp != ".", ]
ALDH5A1 <- ALDH5A1[ALDH5A1$snp != ".", ]; ALDH6A1 <- ALDH6A1[ALDH6A1$snp != ".", ]
ALDH7A1 <- ALDH7A1[ALDH7A1$snp != ".", ]
```

Extracting eQTL SNPs impacting *ADH* and *ALDH* gene expression.

```{r eqtl_snps}
# 6 ADH genes
ADH1A_esnps <- unique(ADH1A$snp); ADH1B_esnps <- unique(ADH1B$snp)
ADH1C_esnps <- unique(ADH1C$snp); ADH4_esnps <- unique(ADH4$snp)
ADH5_esnps <- unique(ADH5$snp); ADH6_esnps <- unique(ADH6$snp)
# 17 ALDH genes
ALDH16A1_esnps <- unique(ALDH16A1$snp); ALDH18A1_esnps <- unique(ALDH18A1$snp)
ALDH1A1_esnps <- unique(ALDH1A1$snp); ALDH1A2_esnps <- unique(ALDH1A2$snp)
ALDH1A3_esnps <- unique(ALDH1A3$snp); ALDH1B1_esnps <- unique(ALDH1B1$snp)
ALDH1L1_esnps <- unique(ALDH1L1$snp); ALDH1L2_esnps <- unique(ALDH1L2$snp)
ALDH2_esnps <- unique(ALDH2$snp); ALDH3A1_esnps <- unique(ALDH3A1$snp)
ALDH3A2_esnps <- unique(ALDH3A2$snp); ALDH3B1_esnps <- unique(ALDH3B1$snp)
ALDH3B2_esnps <- unique(ALDH3B2$snp); ALDH4A1_esnps <- unique(ALDH4A1$snp)
ALDH5A1_esnps <- unique(ALDH5A1$snp); ALDH6A1_esnps <- unique(ALDH6A1$snp)
ALDH7A1_esnps <- unique(ALDH7A1$snp)
```

Extracting affected tissues.

```{r affected_tissues}
# 6 ADH genes
ADH1A_tissues <- unique(ADH1A$tissue); ADH1B_tissues <- unique(ADH1B$tissue)
ADH1C_tissues <- unique(ADH1C$tissue); ADH4_tissues <- unique(ADH4$tissue)
ADH5_tissues <- unique(ADH5$tissue); ADH6_tissues <- unique(ADH6$tissue)
# 17 ALDH genes
ALDH16A1_tissues <- unique(ALDH16A1$tissue); ALDH18A1_tissues <- unique(ALDH18A1$tissue)
ALDH1A1_tissues <- unique(ALDH1A1$tissue); ALDH1A2_tissues <- unique(ALDH1A2$tissue)
ALDH1A3_tissues <- unique(ALDH1A3$tissue); ALDH1B1_tissues <- unique(ALDH1B1$tissue)
ALDH1L1_tissues <- unique(ALDH1L1$tissue); ALDH1L2_tissues <- unique(ALDH1L2$tissue)
ALDH2_tissues <- unique(ALDH2$tissue); ALDH3A1_tissues <- unique(ALDH3A1$tissue)
ALDH3A2_tissues <- unique(ALDH3A2$tissue); ALDH3B1_tissues <- unique(ALDH3B1$tissue)
ALDH3B2_tissues <- unique(ALDH3B2$tissue); ALDH4A1_tissues <- unique(ALDH4A1$tissue)
ALDH5A1_tissues <- unique(ALDH5A1$tissue); ALDH6A1_tissues <- unique(ALDH6A1$tissue)
ALDH7A1_tissues <- unique(ALDH7A1$tissue)
```

### 2. *ADH* and *ALDH* gene expression across GTEx tissues.

```{r gtex_gene_expression, fig.width=12, fig.height=10}
# getting gene expression for all genes across all gtex tissues
gtex <- gzfile("data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz",'rt')  
gtex_expr <- read.table(gtex, skip = 2, sep = "\t", header = TRUE)
adh_aldh_genes <- c("ADH1A", "ADH1B", "ADH1C", "ADH4", "ADH5", "ADH6", "ALDH16A1", "ALDH18A1",
                    "ALDH1A1", "ALDH1A2", "ALDH1A3", "ALDH1B1", "ALDH1L1", "ALDH1L2", "ALDH2",
                    "ALDH3A1", "ALDH3A2", "ALDH3B1", "ALDH3B2", "ALDH4A1", "ALDH5A1", "ALDH6A1",
                    "ALDH7A1")
adh_aldh_expr <- gtex_expr[gtex_expr$Description %in% adh_aldh_genes, ]

# fix column names
colnames(adh_aldh_expr) <- sub("[.]$", "", colnames(adh_aldh_expr)) # remove dot in the end
colnames(adh_aldh_expr) <- gsub("\\.{1,}", "_", colnames(adh_aldh_expr)) # replace any dot with underscore

# Heatmap of log10(TPM + 1) gene expression across 49 GTEx tissues
#adh_aldh_expr_melt <- melt(adh_aldh_expr, id="Description", measure.vars = colnames(adh_aldh_expr[3:56])); colnames(adh_aldh_expr_melt) <- c("name", "tissue", "tpm")
#adh_aldh_expr_melt$log10tpm <- log10(adh_aldh_expr_melt$tpm + 1) # transform to log10(TPM + 1)

adh_aldh_expr[,3:56] <- log10(adh_aldh_expr[,3:56]+1)
adh_aldh_expr_sub <- adh_aldh_expr[2:56]
adh_aldh_expr_sub <- remove_rownames(adh_aldh_expr_sub) %>% column_to_rownames(var="Description")
adh_aldh_mat <- as.matrix(adh_aldh_expr_sub) # convert to matrix

colors = colorRamp2(c(0, max(adh_aldh_mat)), c("white","black"), space = "RGB")

#pdf("figures/adh_aldh_gene_expression_heatmap.pdf", width = 16, height = 16)
ht_ge <- Heatmap(t(adh_aldh_mat), name = "log10(TPM+1)",
                 col = colors, na_col = "white",
                 cluster_rows = FALSE, cluster_columns = FALSE, row_names_side = "left",
                 column_names_side = "top", column_names_gp = gpar(fontface = "italic"),
                 column_order = adh_aldh_genes,
                 width = unit(14, "cm"), height = unit(22, "cm"),
                 border = TRUE, rect_gp = gpar(col = "grey"),
                 heatmap_legend_param = list(legend_direction = "horizontal",
                                             legend_width = unit(5, "cm")))
draw(ht_ge, heatmap_legend_side = "bottom")
#dev.off()
```

### 3. *ADH* and *ALDH* gene regulation across GTEx tissues.

```{r gene_regulation, fig.width=12, fig.height=10}
# This function checks up- and downregulation of ADH and ALDH genes in GTEx tissues. It takes codes3d input as a dataframe and returns if the gene is not expressed, up-, down-, or up/downregulated
all_t <- rownames(t(adh_aldh_mat))
up_down <- function(x){
    ts <- unique(x$tissue) # vector of tissues
    new <- c()
    for (t in 1:length(all_t)){
        if (all_t[t] %in% ts){
            afs <- x[x$tissue==ts[t],]$log2_aFC # vector of effect sizes
            if (all(na.omit(afs) > 0)){
                new <- c(new, "up")
            } else if (all(na.omit(afs) < 0)) {
                new <- c(new, "down")
            } else if (all(na.omit(afs) == 0)) {
                new <- c(new, NA)
            } else {
                new <- c(new, "both")
            }
        } else{
            new <- c(new, NA)
        }
    }
    new
}

# create a dataframe and populate it with up- and downregulation values
df <- data.frame(ADH1A = up_down(ADH1A), ADH1B = up_down(ADH1B), ADH1C = up_down(ADH1C),
                 ADH4 = up_down(ADH4), ADH5 = up_down(ADH5), ADH6= up_down(ADH6),
                 ALDH16A1 = up_down(ALDH16A1), ALDH18A1 = up_down(ALDH18A1),
                 ALDH1A1 = up_down(ALDH1A1), ALDH1A2 = up_down(ALDH1A2),
                 ALDH1A3 = up_down(ALDH1A3), ALDH1B1 = up_down(ALDH1B1),
                 ALDH1L1 = up_down(ALDH1L1), ALDH1L2 = up_down(ALDH1L2), ALDH2 = up_down(ALDH2),
                 ALDH3A1 = up_down(ALDH3A1), ALDH3A2 = up_down(ALDH3A2),
                 ALDH3B1 = up_down(ALDH3B1), ALDH3B2 = up_down(ALDH3B2),
                 ALDH4A1 = up_down(ALDH4A1), ALDH5A1 = up_down(ALDH5A1),
                 ALDH6A1 = up_down(ALDH6A1), ALDH7A1 = up_down(ALDH7A1),
                 row.names = rownames(t(adh_aldh_mat)))

discrete_mat <- as.matrix(df)
colors = structure(c("#ED1C24", "#1C75BC", "black"), names = c("up", "down", "both"))

#pdf("figures/adh_aldh_gene_regulation_heatmap.pdf", width = 16, height = 16)
ht_gr <- Heatmap(discrete_mat, name = "regulation",
                 col = colors, na_col = "white",
                 cluster_rows = FALSE, cluster_columns = FALSE, row_names_side = "left",
                 column_names_side = "top", column_names_gp = gpar(fontface = "italic"),
                 column_order = adh_aldh_genes,
                 width = unit(14, "cm"), height = unit(22, "cm"),
                 border = TRUE, rect_gp = gpar(col = "grey"),
                 heatmap_legend_param = list(legend_direction = "horizontal",
                                             legend_width = unit(5, "cm")))
draw(ht_gr, heatmap_legend_side = "bottom")
#dev.off()
```

### 4. Association analysis of eQTLs involved in *ADH* and *ALDH* gene regulation.

To determine if *ADH* and *ALDH* gene eQTLs are associated with GWAS traits we used LDtraitR package. Register [here](https://ldlink.nci.nih.gov/?tab=apiaccess) to access LDtrait API with R. You'll get your own token. In the function below specify your own token. This analysis was done on 05/02/2021. GWAS Catalog last updated on 05/02/2021, 07:05 PM (GMT+1300).

```{r LDtrait}
# This function queriies LDtrait. It takes a vector of eQTLs, splits the vector into list of vectors of length 50 (or less) and sends a request for each of these vectors. It outputs the dataframe with the query results for all eQTLs.
query_LDtrait <- function(esnps){
    df <- data.frame()
    max <- 50
    ind <- seq_along(esnps)
    esnps_splits <- split(esnps, ceiling(ind/max))
    cat("Total number of SNP chunks to analyse: ", length(esnps_splits), "\n")
    for(i in 1:length(esnps_splits)){
        #skip_to_next <- FALSE
        cat("Analysing SNP chunk: ", i, "\n")
        tryCatch({
            snps <- unlist(esnps_splits[i], use.names=FALSE)
            t <- LDtrait(snps = snps, pop = "CEU", r2d = "r2", r2d_threshold = "0.8", 
                     win_size = "500000", token = "5108b14fedb3")
            df <- rbind(df, t)
        }, error=function(e){
            cat("ERROR: ", conditionMessage(e), "\n")
            #skip_to_next <<- TRUE
        })
        #if(skip_to_next){ next }
    }
    return(df)
}

## 6 ADH genes
ADH1A_df <- query_LDtrait(ADH1A_esnps) # all associations
ADH1A_df$P_value <- as.numeric(ADH1A_df$P_value)
#write.table(ADH1A_df, file = "results/LD/ADH1A_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH1A_df_gw <- subset(ADH1A_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH1A_df_gw, file = "results/LD/ADH1A_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ADH1B_df <- query_LDtrait(ADH1B_esnps) # all associations
ADH1B_df$P_value <- as.numeric(ADH1B_df$P_value)
#write.table(ADH1B_df, file = "results/LD/ADH1B_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH1B_df_gw <- subset(ADH1B_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH1B_df_gw, file = "results/LD/ADH1B_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ADH1C_df <- query_LDtrait(ADH1C_esnps) # all associations
ADH1C_df$P_value <- as.numeric(ADH1C_df$P_value)
#write.table(ADH1C_df, file = "results/LD/ADH1C_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH1C_df_gw <- subset(ADH1C_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH1C_df_gw, file = "results/LD/ADH1C_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ADH4_df <- query_LDtrait(ADH4_esnps) # all associations
ADH4_df$P_value <- as.numeric(ADH4_df$P_value)
#write.table(ADH4_df, file = "results/LD/ADH4_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH4_df_gw <- subset(ADH4_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH4_df_gw, file = "results/LD/ADH4_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ADH5_df <- query_LDtrait(ADH5_esnps) # all associations
ADH5_df$P_value <- as.numeric(ADH5_df$P_value)
#write.table(ADH5_df, file = "results/LD/ADH5_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH5_df_gw <- subset(ADH5_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH5_df_gw, file = "results/LD/ADH5_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ADH6_df <- query_LDtrait(ADH6_esnps) # all associations
ADH6_df$P_value <- as.numeric(ADH6_df$P_value)
#write.table(ADH6_df, file = "results/LD/ADH6_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ADH6_df_gw <- subset(ADH6_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ADH6_df_gw, file = "results/LD/ADH6_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

## 17 ALDH genes
ALDH16A1_df <- query_LDtrait(ALDH16A1_esnps) # all associations
ALDH16A1_df$P_value <- as.numeric(ALDH16A1_df$P_value)
#write.table(ALDH16A1_df, file = "results/LD/ALDH16A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH16A1_df_gw <- subset(ALDH16A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH16A1_df_gw, file = "results/LD/ALDH16A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH18A1_df <- query_LDtrait(ALDH18A1_esnps) # all associations
ALDH18A1_df$P_value <- as.numeric(ALDH18A1_df$P_value)
#write.table(ALDH18A1_df, file = "results/LD/ALDH18A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH18A1_df_gw <- subset(ALDH18A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH18A1_df_gw, file = "results/LD/ALDH18A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1A1_df <- query_LDtrait(ALDH1A1_esnps) # all associations
ALDH1A1_df$P_value <- as.numeric(ALDH1A1_df$P_value)
#write.table(ALDH1A1_df, file = "results/LD/ALDH1A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1A1_df_gw <- subset(ALDH1A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1A1_df_gw, file = "results/LD/ALDH1A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1A2_df <- query_LDtrait(ALDH1A2_esnps) # all associations
ALDH1A2_df$P_value <- as.numeric(ALDH1A2_df$P_value)
#write.table(ALDH1A2_df, file = "results/LD/ALDH1A2_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1A2_df_gw <- subset(ALDH1A2_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1A2_df_gw, file = "results/LD/ALDH1A2_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1A3_df <- query_LDtrait(ALDH1A3_esnps) # all associations
ALDH1A3_df$P_value <- as.numeric(ALDH1A3_df$P_value)
#write.table(ALDH1A3_df, file = "results/LD/ALDH1A3_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1A3_df_gw <- subset(ALDH1A3_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1A3_df_gw, file = "results/LD/ALDH1A3_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1B1_df <- query_LDtrait(ALDH1B1_esnps) # all associations
ALDH1B1_df$P_value <- as.numeric(ALDH1B1_df$P_value)
#write.table(ALDH1B1_df, file = "results/LD/ALDH1B1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1B1_df_gw <- subset(ALDH1B1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1B1_df_gw, file = "results/LD/ALDH1B1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1L1_df <- query_LDtrait(ALDH1L1_esnps) # all associations
ALDH1L1_df$P_value <- as.numeric(ALDH1L1_df$P_value)
#write.table(ALDH1L1_df, file = "results/LD/ALDH1L1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1L1_df_gw <- subset(ALDH1L1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1L1_df_gw, file = "results/LD/ALDH1L1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH1L2_df <- query_LDtrait(ALDH1L2_esnps) # all associations
ALDH1L2_df$P_value <- as.numeric(ALDH1L2_df$P_value)
#write.table(ALDH1L2_df, file = "results/LD/ALDH1L2_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH1L2_df_gw <- subset(ALDH1L2_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH1L2_df_gw, file = "results/LD/ALDH1L2_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH2_df <- query_LDtrait(ALDH2_esnps) # all associations
ALDH2_df$P_value <- as.numeric(ALDH2_df$P_value)
#write.table(ALDH2_df, file = "results/LD/ALDH2_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH2_df_gw <- subset(ALDH2_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH2_df_gw, file = "results/LD/ALDH2_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH3A1_df <- query_LDtrait(ALDH3A1_esnps) # all associations
ALDH3A1_df$P_value <- as.numeric(ALDH3A1_df$P_value)
#write.table(ALDH3A1_df, file = "results/LD/ALDH3A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH3A1_df_gw <- subset(ALDH3A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH3A1_df_gw, file = "results/LD/ALDH3A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH3A2_df <- query_LDtrait(ALDH3A2_esnps) # all associations
ALDH3A2_df$P_value <- as.numeric(ALDH3A2_df$P_value)
#write.table(ALDH3A2_df, file = "results/LD/ALDH3A2_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH3A2_df_gw <- subset(ALDH3A2_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH3A2_df_gw, file = "results/LD/ALDH3A2_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH3B1_df <- query_LDtrait(ALDH3B1_esnps) # all associations
ALDH3B1_df$P_value <- as.numeric(ALDH3B1_df$P_value)
#write.table(ALDH3B1_df, file = "results/LD/ALDH3B1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH3B1_df_gw <- subset(ALDH3B1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH3B1_df_gw, file = "results/LD/ALDH3B1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH3B2_df <- query_LDtrait(ALDH3B2_esnps) # all associations
ALDH3B2_df$P_value <- as.numeric(ALDH3B2_df$P_value)
#write.table(ALDH3B2_df, file = "results/LD/ALDH3B2_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH3B2_df_gw <- subset(ALDH3B2_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH3B2_df_gw, file = "results/LD/ALDH3B2_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH4A1_df <- query_LDtrait(ALDH4A1_esnps) # all associations
ALDH4A1_df$P_value <- as.numeric(ALDH4A1_df$P_value)
#write.table(ALDH4A1_df, file = "results/LD/ALDH4A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH4A1_df_gw <- subset(ALDH4A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH4A1_df_gw, file = "results/LD/ALDH4A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH5A1_df <- query_LDtrait(ALDH5A1_esnps) # all associations
ALDH5A1_df$P_value <- as.numeric(ALDH5A1_df$P_value)
#write.table(ALDH5A1_df, file = "results/LD/ALDH5A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH5A1_df_gw <- subset(ALDH5A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH5A1_df_gw, file = "results/LD/ALDH5A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH6A1_df <- query_LDtrait(ALDH6A1_esnps) # all associations
ALDH6A1_df$P_value <- as.numeric(ALDH6A1_df$P_value)
#write.table(ALDH6A1_df, file = "results/LD/ALDH6A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH6A1_df_gw <- subset(ALDH6A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH6A1_df_gw, file = "results/LD/ALDH6A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)

ALDH7A1_df <- query_LDtrait(ALDH7A1_esnps) # all associations
ALDH7A1_df$P_value <- as.numeric(ALDH7A1_df$P_value)
#write.table(ALDH7A1_df, file = "results/LD/ALDH7A1_LDtraits.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
ALDH7A1_df_gw <- subset(ALDH7A1_df, P_value < 0.00000005) # only genome-wide significant associations
#write.table(ALDH7A1_df_gw, file = "results/LD/ALDH7A1_LDtraits_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE)
```

Here, we plot percent of all and genome-wide significant associations of *ADH* and *ALDH* eQTLs with GWAS traits.

```{r eQTL-trait_associations}
# to get number of ADH1A eQTLs associated with traits: length(unique(ADH1A_df$Query))
# to get number of ADH1A eQTLs: length(unique(ADH1A_esnps))
# 6 ADH genes
# all associations
adh_order <- c('ADH1A', 'ADH1B', 'ADH1C', 'ADH4', 'ADH5', 'ADH6')
ADH_esnps <- c(length(unique(ADH1A_esnps)), length(unique(ADH1B_esnps)),
               length(unique(ADH1C_esnps)), length(unique(ADH4_esnps)),
               length(unique(ADH5_esnps)), length(unique(ADH6_esnps)))
ADH_esnps_assoc <- c(length(unique(ADH1A_df$Query)), length(unique(ADH1B_df$Query)),
                     length(unique(ADH1C_df$Query)), length(unique(ADH4_df$Query)),
                     length(unique(ADH5_df$Query)), length(unique(ADH6_df$Query)))

ADH.df <- data.frame(
    gene = rep(adh_order, 2),
    esnps = rep(c("with association","without association"), each=6),
    number = c(ADH_esnps_assoc, ADH_esnps-ADH_esnps_assoc),
    percentage = c(round(ADH_esnps_assoc/ADH_esnps*100, 2),
                   100-round(ADH_esnps_assoc/ADH_esnps*100, 2)))

#tiff("figures/percent_ADH_eQTLs.tif", units="in", width=10, height=8, res=300)
pdf("figures/percent_ADH_eQTLs.pdf", width = 10, height = 8)
#jpeg("figures/percent_ADH_eQTLs.jpeg", units="in", width=10, height=8, res=300)
ggplot(ADH.df, aes(x = gene, y = percentage, fill = esnps)) +
    geom_bar(stat="identity", position = "stack") + 
    theme_classic() +
    theme(plot.title = element_blank(),
        axis.title.x = element_blank(),
        legend.text=element_text(size=20),
        legend.title=element_blank(),
        legend.position = "bottom",
        legend.direction = "horizontal",
        axis.text=element_text(size=20, colour = "black"),
        axis.text.x=element_text(face = "italic", angle = 45, hjust=1),
        axis.title=element_text(size=24, colour = "black"),
        strip.text.x = element_text(size = 19, colour = "black")) +
    geom_text(aes(y = percentage, label = paste0(number)),
            position=position_stack(vjust = 0.5), vjust=-0.25, color = "black", 
            size = 8, fontface = 'italic') +
    labs(y = "Percentage")
dev.off()

# genome-wide significant associations
ADH_esnps_assoc_gw <- c(length(unique(ADH1A_df_gw$Query)), length(unique(ADH1B_df_gw$Query)),
                     length(unique(ADH1C_df_gw$Query)), length(unique(ADH4_df_gw$Query)),
                     length(unique(ADH5_df_gw$Query)), length(unique(ADH6_df_gw$Query)))

ADH_gw.df <- data.frame(
  gene = adh_order,
  number = ADH_esnps_assoc_gw,
  percentage = c(round(ADH_esnps_assoc_gw/ADH_esnps_assoc*100, 2)))

#tiff("figures/percent_ADH_eQTLs_gw.tif", units="in", width=10, height=8, res=300)
pdf("figures/percent_ADH_eQTLs_gw.pdf", width = 10, height = 8)
#jpeg("figures/percent_ADH_eQTLs_gw.jpeg", units="in", width=10, height=8, res=300)
ggplot(ADH_gw.df, aes(x = gene, y = percentage)) +
    geom_bar(stat="identity", position = "identity") + 
    theme_classic() +
    theme(plot.title = element_blank(),
        axis.title.x = element_blank(),
        legend.text=element_text(size=20),
        legend.title=element_blank(),
        legend.position = "bottom",
        legend.direction = "horizontal",
        axis.text=element_text(size=20, colour = "black"),
        axis.text.x=element_text(face = "italic", angle = 45, hjust=1),
        axis.title=element_text(size=24, colour = "black"),
        strip.text.x = element_text(size = 19, colour = "black")) +
    geom_text(aes(y = percentage, label = paste0(number)),
            position=position_dodge(width=0.9), vjust=-0.25, color = "black", 
            size = 8, fontface = 'italic') +
    labs(y = "Percentage")
dev.off()

# 17 ALDH genes
# all associations
aldh_order <- c('ALDH16A1', 'ALDH18A1', 'ALDH1A1', 'ALDH1A2', 'ALDH1A3', 'ALDH1B1', 'ALDH1L1',
                'ALDH1L2', 'ALDH2', 'ALDH3A1', 'ALDH3A2', 'ALDH3B1', 'ALDH3B2', 'ALDH4A1', 'ALDH5A1',
                'ALDH6A1', 'ALDH7A1')
ALDH_esnps <- c(length(unique(ALDH16A1_esnps)), length(unique(ALDH18A1_esnps)),
               length(unique(ALDH1A1_esnps)), length(unique(ALDH1A2_esnps)),
               length(unique(ALDH1A3_esnps)), length(unique(ALDH1B1_esnps)),
               length(unique(ALDH1L1_esnps)), length(unique(ALDH1L2_esnps)),
               length(unique(ALDH2_esnps)), length(unique(ALDH3A1_esnps)),
               length(unique(ALDH3A2_esnps)), length(unique(ALDH3B1_esnps)),
               length(unique(ALDH3B2_esnps)), length(unique(ALDH4A1_esnps)),
               length(unique(ALDH5A1_esnps)), length(unique(ALDH6A1_esnps)),
               length(unique(ALDH7A1_esnps)))
ALDH_esnps_assoc <- c(length(unique(ALDH16A1_df$Query)), length(unique(ALDH18A1_df$Query)),
                      length(unique(ALDH1A1_df$Query)), 0, 0, length(unique(ALDH1B1_df$Query)),
                      length(unique(ALDH1L1_df$Query)), length(unique(ALDH1L2_df$Query)),
                      length(unique(ALDH2_df$Query)), length(unique(ALDH3A1_df$Query)),
                      length(unique(ALDH3A2_df$Query)), length(unique(ALDH3B1_df$Query)),
                      length(unique(ALDH3B2_df$Query)), 0, length(unique(ALDH5A1_df$Query)),
                      length(unique(ALDH6A1_df$Query)), length(unique(ALDH7A1_df$Query)))

ALDH.df <- data.frame(
    gene = rep(aldh_order, 2),
    esnps = rep(c("with association","without association"), each=17),
    number = c(ALDH_esnps_assoc, ALDH_esnps-ALDH_esnps_assoc),
    percentage = c(round(ALDH_esnps_assoc/ALDH_esnps*100, 2),
                   100-round(ALDH_esnps_assoc/ALDH_esnps*100, 2)))

#tiff("figures/percent_ALDH_eQTLs.tif", units="in", width=20, height=8, res=300)
pdf("figures/percent_ALDH_eQTLs.pdf", width = 20, height = 8)
#jpeg("figures/percent_ALDH_eQTLs.jpeg", units="in", width=20, height=8, res=300)
ggplot(ALDH.df, aes(x = gene, y = percentage, fill = esnps)) +
    geom_bar(stat="identity", position = "stack") + 
    theme_classic() +
    theme(plot.title = element_blank(),
        axis.title.x = element_blank(),
        legend.text=element_text(size=20),
        legend.title=element_blank(),
        legend.position = "bottom",
        legend.direction = "horizontal",
        axis.text=element_text(size=20, colour = "black"),
        axis.text.x=element_text(face = "italic", angle = 45, hjust=1),
        axis.title=element_text(size=24, colour = "black"),
        strip.text.x = element_text(size = 19, colour = "black")) +
    geom_text(aes(y = percentage, label = paste0(number)),
            position=position_stack(vjust = 0.4), vjust=-0.25, color = "black", 
            size = 8, fontface = 'italic') +
    labs(y = "Percentage")
dev.off()

# genome-wide significant associations
ALDH_esnps_assoc_gw <- c(length(unique(ALDH16A1_df_gw$Query)), length(unique(ALDH18A1_df_gw$Query)),
                      length(unique(ALDH1A1_df_gw$Query)), 0, 0, length(unique(ALDH1B1_df_gw$Query)),
                      length(unique(ALDH1L1_df_gw$Query)), length(unique(ALDH1L2_df_gw$Query)),
                      length(unique(ALDH2_df_gw$Query)), length(unique(ALDH3A1_df_gw$Query)),
                      length(unique(ALDH3A2_df_gw$Query)), length(unique(ALDH3B1_df_gw$Query)),
                      length(unique(ALDH3B2_df_gw$Query)), 0, length(unique(ALDH5A1_df_gw$Query)),
                      length(unique(ALDH6A1_df_gw$Query)), length(unique(ALDH7A1_df_gw$Query)))

ALDH_gw.df <- data.frame(
    gene = aldh_order,
    number = ALDH_esnps_assoc_gw,
    percentage = c(round(ALDH_esnps_assoc_gw/ALDH_esnps_assoc*100, 2)))

#tiff("figures/percent_ALDH_eQTLs_gw.tif", units="in", width=20, height=8, res=300)
pdf("figures/percent_ALDH_eQTLs_gw.pdf", width = 20, height = 8)
#jpeg("figures/percent_ADH_eQTLs_gw.jpeg", units="in", width=20, height=8, res=300)
ggplot(ALDH_gw.df, aes(x = gene, y = percentage)) +
    geom_bar(stat="identity", position = "identity") + 
    theme_classic() +
    theme(plot.title = element_blank(),
        axis.title.x = element_blank(),
        legend.text=element_text(size=20),
        legend.title=element_blank(),
        legend.position = "bottom",
        legend.direction = "horizontal",
        axis.text=element_text(size=20, colour = "black"),
        axis.text.x=element_text(face = "italic", angle = 45, hjust=1),
        axis.title=element_text(size=24, colour = "black"),
        strip.text.x = element_text(size = 19, colour = "black")) +
    geom_text(aes(y = percentage, label = paste0(number)),
            position=position_dodge(width=0.9), vjust=-0.25, color = "black", 
            size = 8, fontface = 'italic') +
    labs(y = "Percentage")
dev.off()
```





