---
title: "ADH and ALDH gene regulation"
author: "Evgeniia Golovina"
date: "03/02/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# install.packages("pacman")
# load libraries
pacman::p_load(ggplot2, ComplexHeatmap, dplyr, tidyr, reshape2, tidyverse, circlize)

setwd("/Users/foffa/Desktop/ADH_and_ALDH_regulatory_networks")
```

## Reporducibility report for "*ADH* and *ALDH* gene regulation" study.

This R Markdown document generates the reproducibility report for "*ADH* and *ALDH* gene regulation" study. 

### 1. CoDeS3D analysis

First, we run CoDeS3D pipeline (gene mode, GTEx eqtl project) to get regulatory interactions for each of *ADH* and *ALDH* gene across all GTEx (v8) tissues: `python codes3d/codes3d.py -g ALDH1A1 -o results/codes3d/ALDH1A1 --eqtl-project GTEx`. 

```{r loading_codes3d_results}
# 6 ADH genes; there were no significant interactions for ADH7 gene
ADH1A <- read.table("results/codes3d/ADH1A/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH1B <- read.table("results/codes3d/ADH1B/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH1C <- read.table("results/codes3d/ADH1C/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH4 <- read.table("results/codes3d/ADH4/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH5 <- read.table("results/codes3d/ADH5/significant_eqtls.txt", header = TRUE, sep = "\t")
ADH6 <- read.table("results/codes3d/ADH6/significant_eqtls.txt", header = TRUE, sep = "\t")
# 17 ALDH genes; there were no ignificant interactions for ALDH8A1 and ALDH9A1 genes
ALDH16A1 <- read.table("results/codes3d/ALDH16A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH18A1 <- read.table("results/codes3d/ALDH18A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A1 <- read.table("results/codes3d/ALDH1A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A2 <- read.table("results/codes3d/ALDH1A2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1A3 <- read.table("results/codes3d/ALDH1A3/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1B1 <- read.table("results/codes3d/ALDH1B1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1L1 <- read.table("results/codes3d/ALDH1L1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH1L2 <- read.table("results/codes3d/ALDH1L2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH2 <- read.table("results/codes3d/ALDH2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3A1 <- read.table("results/codes3d/ALDH3A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3A2 <- read.table("results/codes3d/ALDH3A2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3B1 <- read.table("results/codes3d/ALDH3B1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH3B2 <- read.table("results/codes3d/ALDH3B2/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH4A1 <- read.table("results/codes3d/ALDH4A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH5A1 <- read.table("results/codes3d/ALDH5A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH6A1 <- read.table("results/codes3d/ALDH6A1/significant_eqtls.txt", header = TRUE, sep = "\t")
ALDH7A1 <- read.table("results/codes3d/ALDH7A1/significant_eqtls.txt", header = TRUE, sep = "\t")

# remove interactions with missing rsID number for snp
# 6 ADH genes
ADH1A <- ADH1A[ADH1A$snp != ".", ]; ADH1B <- ADH1B[ADH1B$snp != ".", ]
ADH1C <- ADH1C[ADH1C$snp != ".", ]; ADH4 <- ADH4[ADH4$snp != ".", ]
ADH5 <- ADH5[ADH5$snp != ".", ]; ADH6 <- ADH6[ADH6$snp != ".", ]
# 17 ALDH genes
ALDH16A1 <- ALDH16A1[ALDH16A1$snp != ".", ]; ALDH18A1 <- ALDH18A1[ALDH18A1$snp != ".", ]
ALDH1A1 <- ALDH1A1[ALDH1A1$snp != ".", ]; ALDH1A2 <- ALDH1A2[ALDH1A2$snp != ".", ]
ALDH1A3 <- ALDH1A3[ALDH1A3$snp != ".", ]; ALDH1B1 <- ALDH1B1[ALDH1B1$snp != ".", ]
ALDH1L1 <- ALDH1L1[ALDH1L1$snp != ".", ]; ALDH1L2 <- ALDH1L2[ALDH1L2$snp != ".", ]
ALDH2 <- ALDH2[ALDH2$snp != ".", ]; ALDH3A1 <- ALDH3A1[ALDH3A1$snp != ".", ]
ALDH3A2 <- ALDH3A2[ALDH3A2$snp != ".", ]; ALDH3B1 <- ALDH3B1[ALDH3B1$snp != ".", ]
ALDH3B2 <- ALDH3B2[ALDH3B2$snp != ".", ]; ALDH4A1 <- ALDH4A1[ALDH4A1$snp != ".", ]
ALDH5A1 <- ALDH5A1[ALDH5A1$snp != ".", ]; ALDH6A1 <- ALDH6A1[ALDH6A1$snp != ".", ]
ALDH7A1 <- ALDH7A1[ALDH7A1$snp != ".", ]
```

Extracting eQTL SNPs impacting *ADH* and *ALDH* gene expression.

```{r eqtl_snps}
# 6 ADH genes
ADH1A_esnps <- unique(ADH1A$snp); ADH1B_esnps <- unique(ADH1B$snp)
ADH1C_esnps <- unique(ADH1C$snp); ADH4_esnps <- unique(ADH4$snp)
ADH5_esnps <- unique(ADH5$snp); ADH6_esnps <- unique(ADH6$snp)
# 17 ALDH genes
ALDH16A1_esnps <- unique(ALDH16A1$snp); ALDH18A1_esnps <- unique(ALDH18A1$snp)
ALDH1A1_esnps <- unique(ALDH1A1$snp); ALDH1A2_esnps <- unique(ALDH1A2$snp)
ALDH1A3_esnps <- unique(ALDH1A3$snp); ALDH1B1_esnps <- unique(ALDH1B1$snp)
ALDH1L1_esnps <- unique(ALDH1L1$snp); ALDH1L2_esnps <- unique(ALDH1L2$snp)
ALDH2_esnps <- unique(ALDH2$snp); ALDH3A1_esnps <- unique(ALDH3A1$snp)
ALDH3A2_esnps <- unique(ALDH3A2$snp); ALDH3B1_esnps <- unique(ALDH3B1$snp)
ALDH3B2_esnps <- unique(ALDH3B2$snp); ALDH4A1_esnps <- unique(ALDH4A1$snp)
ALDH5A1_esnps <- unique(ALDH5A1$snp); ALDH6A1_esnps <- unique(ALDH6A1$snp)
ALDH7A1_esnps <- unique(ALDH7A1$snp)
```

Extracting affected tissues.

```{r affected_tissues}
# 6 ADH genes
ADH1A_tissues <- unique(ADH1A$tissue); ADH1B_tissues <- unique(ADH1B$tissue)
ADH1C_tissues <- unique(ADH1C$tissue); ADH4_tissues <- unique(ADH4$tissue)
ADH5_tissues <- unique(ADH5$tissue); ADH6_tissues <- unique(ADH6$tissue)
# 17 ALDH genes
ALDH16A1_tissues <- unique(ALDH16A1$tissue); ALDH18A1_tissues <- unique(ALDH18A1$tissue)
ALDH1A1_tissues <- unique(ALDH1A1$tissue); ALDH1A2_tissues <- unique(ALDH1A2$tissue)
ALDH1A3_tissues <- unique(ALDH1A3$tissue); ALDH1B1_tissues <- unique(ALDH1B1$tissue)
ALDH1L1_tissues <- unique(ALDH1L1$tissue); ALDH1L2_tissues <- unique(ALDH1L2$tissue)
ALDH2_tissues <- unique(ALDH2$tissue); ALDH3A1_tissues <- unique(ALDH3A1$tissue)
ALDH3A2_tissues <- unique(ALDH3A2$tissue); ALDH3B1_tissues <- unique(ALDH3B1$tissue)
ALDH3B2_tissues <- unique(ALDH3B2$tissue); ALDH4A1_tissues <- unique(ALDH4A1$tissue)
ALDH5A1_tissues <- unique(ALDH5A1$tissue); ALDH6A1_tissues <- unique(ALDH6A1$tissue)
ALDH7A1_tissues <- unique(ALDH7A1$tissue)
```

### 2. *ADH* and *ALDH* gene expression across GTEx tissues.

```{r gtex_gene_expression, fig.width=12, fig.height=10}
# getting gene expression for all genes across all gtex tissues
gtex <- gzfile("data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz",'rt')  
gtex_expr <- read.table(gtex, skip = 2, sep = "\t", header = TRUE)
adh_aldh_genes <- c("ADH1A", "ADH1B", "ADH1C", "ADH4", "ADH5", "ADH6", "ALDH16A1", "ALDH18A1",
                    "ALDH1A1", "ALDH1A2", "ALDH1A3", "ALDH1B1", "ALDH1L1", "ALDH1L2", "ALDH2",
                    "ALDH3A1", "ALDH3A2", "ALDH3B1", "ALDH3B2", "ALDH4A1", "ALDH5A1", "ALDH6A1",
                    "ALDH7A1")
adh_aldh_expr <- gtex_expr[gtex_expr$Description %in% adh_aldh_genes, ]

# fix column names
colnames(adh_aldh_expr) <- sub("[.]$", "", colnames(adh_aldh_expr)) # remove dot in the end
colnames(adh_aldh_expr) <- gsub("\\.{1,}", "_", colnames(adh_aldh_expr)) # replace any dot with underscore

# Heatmap of log10(TPM + 1) gene expression across 49 GTEx tissues
#adh_aldh_expr_melt <- melt(adh_aldh_expr, id="Description", measure.vars = colnames(adh_aldh_expr[3:56])); colnames(adh_aldh_expr_melt) <- c("name", "tissue", "tpm")
#adh_aldh_expr_melt$log10tpm <- log10(adh_aldh_expr_melt$tpm + 1) # transform to log10(TPM + 1)

adh_aldh_expr[,3:56] <- log10(adh_aldh_expr[,3:56]+1)
adh_aldh_expr_sub <- adh_aldh_expr[2:56]
adh_aldh_expr_sub <- remove_rownames(adh_aldh_expr_sub) %>% column_to_rownames(var="Description")
adh_aldh_mat <- as.matrix(adh_aldh_expr_sub) # convert to matrix

colors = colorRamp2(c(0, max(adh_aldh_mat)), c("white","black"), space = "RGB")

#pdf("figures/adh_aldh_gene_expression_heatmap.pdf", width = 16, height = 16)
ht_ge <- Heatmap(t(adh_aldh_mat), name = "log10(TPM+1)",
                 col = colors, na_col = "white",
                 cluster_rows = FALSE, cluster_columns = FALSE, row_names_side = "left",
                 column_names_side = "top", column_names_gp = gpar(fontface = "italic"),
                 column_order = adh_aldh_genes,
                 width = unit(14, "cm"), height = unit(22, "cm"),
                 border = TRUE, rect_gp = gpar(col = "grey"),
                 heatmap_legend_param = list(legend_direction = "horizontal",
                                             legend_width = unit(5, "cm")))
draw(ht_ge, heatmap_legend_side = "bottom")
#dev.off()
```

### 3. *ADH* and *ALDH* gene regulation across GTEx tissues.

```{r gene_regulation, fig.width=12, fig.height=10}
# This function checks up- and downregulation of ADH and ALDH genes in GTEx tissues. It takes codes3d input as a dataframe and returns if the gene is not expressed, up-, down-, or up/downregulated
all_t <- rownames(t(adh_aldh_mat))
up_down <- function(x){
    ts <- unique(x$tissue) # vector of tissues
    new <- c()
    for (t in 1:length(all_t)){
        if (all_t[t] %in% ts){
            afs <- x[x$tissue==ts[t],]$log2_aFC # vector of effect sizes
            if (all(na.omit(afs) > 0)){
                new <- c(new, "up")
            } else if (all(na.omit(afs) < 0)) {
                new <- c(new, "down")
            } else if (all(na.omit(afs) == 0)) {
                new <- c(new, NA)
            } else {
                new <- c(new, "both")
            }
        } else{
            new <- c(new, NA)
        }
    }
    new
}

# create a dataframe and populate it with up- and downregulation values
df <- data.frame(ADH1A = up_down(ADH1A), ADH1B = up_down(ADH1B), ADH1C = up_down(ADH1C),
                 ADH4 = up_down(ADH4), ADH5 = up_down(ADH5), ADH6= up_down(ADH6),
                 ALDH16A1 = up_down(ALDH16A1), ALDH18A1 = up_down(ALDH18A1),
                 ALDH1A1 = up_down(ALDH1A1), ALDH1A2 = up_down(ALDH1A2),
                 ALDH1A3 = up_down(ALDH1A3), ALDH1B1 = up_down(ALDH1B1),
                 ALDH1L1 = up_down(ALDH1L1), ALDH1L2 = up_down(ALDH1L2), ALDH2 = up_down(ALDH2),
                 ALDH3A1 = up_down(ALDH3A1), ALDH3A2 = up_down(ALDH3A2),
                 ALDH3B1 = up_down(ALDH3B1), ALDH3B2 = up_down(ALDH3B2),
                 ALDH4A1 = up_down(ALDH4A1), ALDH5A1 = up_down(ALDH5A1),
                 ALDH6A1 = up_down(ALDH6A1), ALDH7A1 = up_down(ALDH7A1),
                 row.names = rownames(t(adh_aldh_mat)))

discrete_mat <- as.matrix(df)
colors = structure(c("#ED1C24", "#1C75BC", "black"), names = c("up", "down", "both"))

#pdf("figures/adh_aldh_gene_regulation_heatmap.pdf", width = 16, height = 16)
ht_gr <- Heatmap(discrete_mat, name = "regulation",
                 col = colors, na_col = "white",
                 cluster_rows = FALSE, cluster_columns = FALSE, row_names_side = "left",
                 column_names_side = "top", column_names_gp = gpar(fontface = "italic"),
                 column_order = adh_aldh_genes,
                 width = unit(14, "cm"), height = unit(22, "cm"),
                 border = TRUE, rect_gp = gpar(col = "grey"),
                 heatmap_legend_param = list(legend_direction = "horizontal",
                                             legend_width = unit(5, "cm")))
draw(ht_gr, heatmap_legend_side = "bottom")
#dev.off()
```

### 4. Association analysis of eQTLs involved in *ADH* and *ALDH* gene regulation.

GWAS SNP-trait associations were downloaded and assessed on 26/08/2020. Only SNPs that have rsIDs were included in the analysis.

```{r gwas_cleaning}
# loading gwas associations
gwas <- gzfile("data/gwas_catalog_v1.0.2-associations_e100_r2020-08-26.tsv.gz",'rt')
gwas_assoc <- read.delim(gwas, header = TRUE, quote= "") # 197,439 associations

# subsetting
#gwas_assoc_sub <- subset(gwas_assoc, select=c(CHR_ID, SNPS, CONTEXT, P.VALUE, MAPPED_TRAIT))
gwas_assoc_sub <- subset(gwas_assoc, select=c(SNPS, P.VALUE, MAPPED_TRAIT))
colnames(gwas_assoc_sub) <- c("snp", "p_value", "mapped_trait")
gwas_assoc_sub <- subset(gwas_assoc_sub, p_value < 0.00000005)
rs <- grep("^rs", gwas_assoc_sub$snp) # removing SNPs that don't have rs IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, snp %in% snp[rs])
x <- grep(" x ", gwas_assoc_sub$snp) # removing SNPs that have "x" in IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[x]))
y <- grep("\\-", gwas_assoc_sub$snp) # removing SNPs that have "-" in IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[y]))
y <- grep("\\_", gwas_assoc_sub$snp) # removing SNPs that have "_" in IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[y]))
y <- grep("\\*", gwas_assoc_sub$snp) # removing SNPs that have "*" in IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[y]))
y <- grep("\\/", gwas_assoc_sub$snp) # removing SNPs that have "/" in IDs
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[y]))
y <- c("rs12524487¬†", "rs2428362‚Ä†")
gwas_assoc_sub <- subset(gwas_assoc_sub, !(snp %in% snp[y])) # removing wierdos

gwas_assoc_uniq <- distinct(gwas_assoc_sub) # getting only unique associations
write.table(gwas_assoc_uniq, file = "results/gwas/gwas_5E-08.txt", sep = "\t", col.names = TRUE, row.names=FALSE) # write them
```

We used `python scripts/fix_snps.py` to split the lines with snps separated by ";". 
We run `python scripts/fix_traits.py` to split the mapped traits by ", " to preserve single snp - single trait relationship.
Next, we run `python scripts/fix_ids.py` to remap merged old rsIDs into a new rsIDs. The  [RsMergeArch.bcp.gz file](ftp://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606_b151_GRCh38p7/database/organism_data/RsMergeArch.bcp.gz) with rs merge table for genome GRCh38p7 build 151 was downloaded from the [dbSNP ftp site](ftp://ftp.ncbi.nlm.nih.gov/snp/organisms/human_9606_b151_GRCh38p7/database/organism_data/) on 27/08/2020.

```{r unique_snp_trait_pairs}
gwas_st_pairs <- read.table("results/gwas/gwas_5E-08_snpsfixed_traitfixed_idfixed.txt", sep = "\t", header = TRUE, quote= "") # 149,096 gwas snp-trait pairs
gwas_st_pairs_uniq <- subset(gwas_st_pairs, select=c(snp.1, mapped_trait)) %>% distinct() # 111,172 unique snp-trait pairs
writeLines(gwas_st_pairs_uniq$snp.1, con = "results/gwas/gwas_5E-08_snps.txt", sep = "\n")
```

To get genomic positions for GWAS SNPs according to genome GRCh38p7 build 151, we downloaded dbSNP bed files for each chromosome from [dbSNP ftp site](ftp://ftp.ncbi.nih.gov/snp/organisms/human_9606_b151_GRCh38p7/BED/), concatenated them and run `bash scripts/rsID2Bed.sh results/gwas/gwas_5E-08_snps.txt`

```{r unique_snp_trait_pairs_with_positions}
colnames(gwas_st_pairs_uniq) <- c("snp", "mapped_trait")
gwas_st_pairs_uniq_pos <- read.table("results/gwas/rsID2Bed/gwas_5E-08_snps.txt.bed", sep = "\t", header = FALSE, quote= "")
colnames(gwas_st_pairs_uniq_pos) <- c("chr", "start", "end", "snp", "score", "strand")
# merge two tables by snp rsID
gwas_st_pairs_uniq_pos <- distinct(gwas_st_pairs_uniq_pos)
merged <- merge(gwas_st_pairs_uniq, gwas_st_pairs_uniq_pos, by = "snp")
```

We overlapped GWAS SNPs with eQTLs to identify how many eQTLs have a direct association with GWAS traits.

```{r gwas_trait_association}
# 6 ADH genes
ADH1A_assoc <- merged[merged$snp %in% ADH1A_esnps, ] # 17 of 586
ADH1B_assoc <- merged[merged$snp %in% ADH1B_esnps, ] # 27 of 533
ADH1C_assoc <- merged[merged$snp %in% ADH1C_esnps, ] # 43 of 850
ADH4_assoc <- merged[merged$snp %in% ADH4_esnps, ] # 66 of 1547
ADH5_assoc <- merged[merged$snp %in% ADH5_esnps, ] # 42 of 1167
ADH6_assoc <- merged[merged$snp %in% ADH6_esnps, ] # 33 of 733
# 17 ALDH genes
ALDH16A1_assoc <- merged[merged$snp %in% ALDH16A1_esnps, ] # 12 of 253
ALDH18A1_assoc <- merged[merged$snp %in% ALDH18A1_esnps, ] # 13 of 751
ALDH1A1_assoc <- merged[merged$snp %in% ALDH1A1_esnps, ] # 0 of 171
ALDH1A2_assoc <- merged[merged$snp %in% ALDH1A2_esnps, ] # 0 of 4
ALDH1A3_assoc <- merged[merged$snp %in% ALDH1A3_esnps, ] # 1 of 12
ALDH1B1_assoc <- merged[merged$snp %in% ALDH1B1_esnps, ] # 1 of 48
ALDH1L1_assoc <- merged[merged$snp %in% ALDH1L1_esnps, ] # 6 of 231
ALDH1L2_assoc <- merged[merged$snp %in% ALDH1L2_esnps, ] # 6 of 595
ALDH2_assoc <- merged[merged$snp %in% ALDH2_esnps, ] # 284 of 1511
ALDH3A1_assoc <- merged[merged$snp %in% ALDH3A1_esnps, ] # 3 of 166
ALDH3A2_assoc <- merged[merged$snp %in% ALDH3A2_esnps, ] # 7 of 276
ALDH3B1_assoc <- merged[merged$snp %in% ALDH3B1_esnps, ] # 0 of 380
ALDH3B2_assoc <- merged[merged$snp %in% ALDH3B2_esnps, ] # 0 of 54
ALDH4A1_assoc <- merged[merged$snp %in% ALDH4A1_esnps, ] # 0 of 6
ALDH5A1_assoc <- merged[merged$snp %in% ALDH5A1_esnps, ] # 11 of 783
ALDH6A1_assoc <- merged[merged$snp %in% ALDH6A1_esnps, ] # 6 of 863
ALDH7A1_assoc <- merged[merged$snp %in% ALDH7A1_esnps, ] # 3 of 446
```

Next, we check the distribution of eQTLs across different chromosomes.

```{r eqtl_chr_distr}


```



LD analysis of *ADH* and *ALDH* gene eQTLs and GWAS SNPs.



